rust/datafusion/src/optimizer/type_coercion.rs

```diff
@@ -183,10 +192,42 @@ mod tests {
     use super::*;
     use crate::execution::context::ExecutionContext;
     use crate::execution::physical_plan::csv::CsvReadOptions;
-    use crate::logicalplan::{col, Operator};
+    use crate::logicalplan::{aggregate_expr, col, lit, Operator};
     use crate::test::arrow_testdata_path;
     use arrow::datatypes::{DataType, Field, Schema};
 
+    #[test]
+    fn test_all_operators() -> Result<()> {
+        let testdata = arrow_testdata_path();
+        let path = format!("{}/csv/aggregate_test_100.csv", testdata);
+
+        let options = CsvReadOptions::new().schema_infer_max_records(100);
+        let plan = LogicalPlanBuilder::scan_csv(&path, options, None)?
+            // filter clause needs the type coercion rule applied
```

Comment:
If you desire to test all operators, as the test is named, perhaps adding `Projection` would also be valuable

Reply:
Thanks. I've added that.